src/libraries/System.Private.CoreLib/src/System/IO/BinaryReader.cs

```diff
@@ -229,24 +220,24 @@ public virtual char ReadChar()
             return (char)value;
         }
 
-        public virtual short ReadInt16() => BinaryPrimitives.ReadInt16LittleEndian(InternalRead(2));
+        public virtual short ReadInt16() => BinaryPrimitives.ReadInt16LittleEndian(InternalRead(stackalloc byte[2]));
```

Comment:
subjective nit: in theory everyone should know the sizes of primitive types, but on the other hand it looks just 100% self-describing when using `typeof(T)` instead consts.

```suggestion
        public virtual short ReadInt16() => BinaryPrimitives.ReadInt16LittleEndian(InternalRead(stackalloc byte[typeof(short)]));
```